// justreddit/build/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.randomSubInfo = exports.subInfo = exports.randomImageFromSub = exports.randomImage = exports.randomImageSub = exports.randomSub = exports.randomPost = exports.randomPostFromSub = void 0;
const randomPostFromSub_1 = require("./functions/randomPostFromSub");
exports.randomPostFromSub = randomPostFromSub_1.default;
const randomPost_1 = require("./functions/randomPost");
exports.randomPost = randomPost_1.default;
const randomSub_1 = require("./functions/randomSub");
exports.randomSub = randomSub_1.default;
const randomImageSub_1 = require("./functions/randomImageSub");
exports.randomImageSub = randomImageSub_1.default;
const randomImage_1 = require("./functions/randomImage");
exports.randomImage = randomImage_1.default;
const randomImageFromSub_1 = require("./functions/randomImageFromSub");
exports.randomImageFromSub = randomImageFromSub_1.default;
const subInfo_1 = require("./functions/subInfo");
exports.subInfo = subInfo_1.default;
const randomSubInfo_1 = require("./functions/randomSubInfo");
exports.randomSubInfo = randomSubInfo_1.default;
__exportStar(require("./types/post"), exports);
__exportStar(require("./types/sub"), exports);
exports.default = randomPost_1.default;
