// justreddit/build/functions/subInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const node_fetch_1 = require("node-fetch");
async function subInfo({ subReddit, excludeRaw = true }) {
    if (subReddit === null)
        return {
            error: "no subreddit",
        };
    const redditFetch = await (0, node_fetch_1.default)(`https://www.reddit.com/r/${subReddit}/about.json`);
    const response = await redditFetch.json();
    if (typeof response.error !== "undefined")
        return {
            error: `reddit error: ${response.error}`,
        };
    const redditInfo = response.data;
    if (redditInfo === null)
        return {
            error: "no subreddit found",
        };
    return {
        subreddit: redditInfo.display_name,
        title: redditInfo.title,
        description: redditInfo.public_description,
        url: `https://www.reddit.com${redditInfo.url}`,
        primaryColor: redditInfo.primary_color,
        keyColor: redditInfo.key_color,
        type: redditInfo.subreddit_type,
        icon: redditInfo.community_icon,
        userCount: redditInfo.subscribers,
        activeUserCount: redditInfo.active_user_count,
        banner: redditInfo.banner_background_image,
        mobileBanner: redditInfo.mobile_banner_image,
        bannerColor: redditInfo.banner_background_color,
        emojisEnabled: redditInfo.emojis_enabled,
        nsfw: redditInfo.over_18,
        createdUTC: redditInfo.created_utc,
        crossPostable: redditInfo.is_crosspostable_subreddit,
        raw: excludeRaw !== true ? redditInfo : null,
    };
}
exports.default = subInfo;
